src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php

```diff
@@ -60,17 +65,24 @@ public function create(string $resourceClass, string $property, array $options =
         }
 
         $propertyMetadata = $this->transformReadWrite($propertyMetadata, $resourceClass, $property, $normalizationGroups, $denormalizationGroups, $ignoredAttributes);
-        $types = $propertyMetadata->getBuiltinTypes() ?? [];
 
-        if (!$this->isResourceClass($resourceClass) && $types) {
-            foreach ($types as $builtinType) {
-                if ($builtinType->isCollection()) {
-                    return $propertyMetadata->withReadableLink(true)->withWritableLink(true);
-                }
+        $type = $propertyMetadata->getPhpType();
+        if ($type && !$this->isResourceClass($resourceClass)) {
+            $typeIsCollection = static function (Type $type) use (&$typeIsCollection): bool {
```

Comment:
this function is often used, maybe that we should create a trait while we wait for helpers inside symfony?